src/anomalib/engine/engine.py

```diff
@@ -349,14 +350,14 @@ def test(
         """
         if model:
             self._setup_trainer(model)
-            self._setup_dataset_task(dataloaders)
+        self._setup_dataset_task(dataloaders)
         return self.trainer.test(model, dataloaders, ckpt_path, verbose, datamodule)
 
     def predict(
         self,
         model: AnomalyModule | None = None,
         dataloaders: EVAL_DATALOADERS | AnomalibDataModule | None = None,
-        datamodule: AnomalibDataModule | None = None,
+        datamodule: AnomalibDataModule | Dataset | InferenceDataset | None = None,
```

Comment:
for a future reference, it would be nice if we use type variables for our main components and store them in like `anomalib.utils.types`

Reply:
Yeah. I think we should refactor all these long types. I've taken a note